robyn/openapi.py

```diff
@@ -277,6 +313,42 @@ def get_openapi_type(self, typed_dict: TypedDict) -> str:
         # default to "string" if type is not found
         return "string"
 
+    def get_properties_object(self, parameter: str, type: Any) -> dict:
+        """
+        Get the properties object for request body
+
+        @param parameter: name of the parameter
+        @param type: Any the type to be inferred
+        @return: dict the properties object
+        """
+
+        properties = {
+            "title": parameter.capitalize(),
+        }
+
+        type_mapping = {
+            int: "integer",
+            str: "string",
+            bool: "boolean",
+            float: "number",
+            dict: "object",
+            list: "array",
+        }
```

Comment:
@VishnuSanal , what happens if it is a custom type?

Reply:
> @VishnuSanal , what happens if it is a custom type?

uh, oh -- it defaults to string at present. any other suggestions? meanwhile I will check what fastapi does & update here.

these are the supported data types: https://swagger.io/specification/#data-types

Reply:
I am not sure. Maybe just use the class name if nothing works. Or recursively parse them?

Reply:
we get the following if we use an "objject" type.

```{
  "name": {},
  "description": "string",
  "price": 0,
  "tax": 0
}```

@sansyrox I think we can default it to it ("object") instead of a string. what say?

Reply:
@VishnuSanal , should recursively parse it like fastpai

Reply:
@VishnuSanal , did you address this?

Reply:
yeps yeps